Analysis #56 Demographic trends

maggiesu0725 opened the pull request.

maggiesu0725 commented:
@rpitingolo please review.

rpitingolo commented:
@maggiesu0725 thanks. I will review now.

rpitingolo commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1036 @@
+by ucounty;
+run;
+
+data populationtrend;
```
In this section you are merging a tract-file with a county file one-to-many. This is causing the 2017 population to be wildly over calculated in the proc summary below.

rpitingolo commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1036 @@
+data vacantunits;
+merge COGSvacant_2017 COGSarea_2017;
+by upuma;
+run;
```
@maggiesu0725 you should do a proc summary to get the number of units and vacant units before merging by upuma. Otherwise if you look at the dataset 'vacancyrate' you'll see it doesn't make sense. Charles county has almost 100% vacancy.

rpitingolo commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1036 @@
+data hh_housing;
+merge COGSarea_2017 vacancyrate;
+by Jurisdiction;
+run;
```
@maggiesu0725 similar issue with this merge. You want to make sure you are doing one-to-one merges rather than one-to-many, otherwise it causes problems later on.

rpitingolo commented:
@maggiesu0725 please review the code specific comments above and make the appropriate edits. I didn't review the ipums code at the very bottom because my computer crashed and I need a break from screen time. I will take another look in a bit.

maggiesu0725 commented:
Thanks, I think I've addressed the two comments. @rpitingolo

maggiesu0725 commented:
@rpitingolo please review the hh type part. Specifically, my categorization for hh types.

rpitingolo commented:
@maggiesu0725 thanks. I am reviewing.

rpitingolo commented:
@maggiesu0725 the data step that creates data hh_housing is still doing a one-to-many merge that's causing the next proc summary to not work as intended

rpitingolo commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1036 @@
+
+else HHcat=4;
+
+retain HHnumber_2000 =1;
```
The merge is not working correctly here because the datasets both need to be sorted by the merge variable. I'm also not sure what the retain code is supposed to do?

maggiesu0725 commented:
Please review when you are back. Had to change HH cat code for 2000 because the hhtype variable is not available. @rpitingolo

rpitingolo commented:
@maggiesu0725 I am reviewing.

rpitingolo commented:
@maggiesu0725 there was a small issue that I corrected and committed. Have you started working on getting this data into the final format? If not I can work on that while you continue working on the other issue.

maggiesu0725 commented:
Thanks, I plan to add the Loudon weights in 2000 and not sure what you mean by final format, but if you could work on that that would be great! @rpitingolo

maggiesu0725 commented:
@ptatian could  you do a final review? The program works but I couldn't figure out why for 2010 the IPUMs data is missing for some Jurisdictions..?

ptatian commented:
@maggiesu0725 I will take a look.

ptatian commented:
@maggiesu0725 I have some feedback for you on this program. Please let me know when we can sit down and chat.

rpitingolo commented:
@maggiesu0725 I'm back at my desk so please let me know if you need help wrapping up this and #58. Thanks.

maggiesu0725 commented:
for population in 70 and 80, census provides by county count here https://www.census.gov/population/www/censusdata/cencounts/index.html but it is in PDF
This is txt version:
https://www.census.gov/data/datasets/time-series/demo/popest/1970s-county.html

Haven't found source for HH counts by county in 1970 yet. 

@ptatian @rpitingolo

lhendey commented:
@maggiesu0725  did you look at NHGIS? @rpitingolo @ptatian

rpitingolo commented:
@lhendey I just suggested NHGIS and am pulling the files from there now. @ptatian @maggiesu0725

rpitingolo commented:
@maggiesu0725 I read in the 1970 and 1980 TotPop and HH data from NHGIS and created a ucounty variable on both. Use the sas datasets saved here: L:\Libraries\RegHsg\Raw\NHGIS

rpitingolo commented:
@maggiesu0725 for some reason the 1970 NHGIS is missing the number of HHs for DC. The number is: 262,538

Source: https://www2.census.gov/prod2/decennial/documents/1970a_dc-01.pdf table 16

ptatian commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1458 @@
+
+data hhestimates17 (drop= _FREQ_);
+set hh_housing;
+hhestimate17= totalunits*(1-vacrate);
```
I think you said that the 2017 population and housing unit counts match the Census estimates, so it doesn't seem like we need this part? And you don't seem to use HHESTIMATE17 anywhere. I would just delete all of this if it is unnecessary.

maggiesu0725 replied:
I used it right in the next step? This dataset uses vacancy rate to calculate the estimated household count that is not in NCDB.

ptatian commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1458 @@
+run;
+
+proc summary data=hhrelate_&year.;
+  class serial;
```
If you use a **class** statement instead of a **by** statement here, put an *nway* option on the proc summary statement. Otherwise, you end up with a total line for all of the people in the data set.

ptatian commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1458 @@
+
+data hhtype_&year.;
+set hhtype_&year.;
+if hhtype in (4,5,6,7) then do;  /*non family*/
```
The coding in this section is wrong. Everyone ends up with HHcat=5. You need some **else if** statements where you are using just **if**'s to fix it.

ptatian replied:
How about this?

‘’’

if hhtype in (4,5,6,7) then do;  /*non family*/
if numprec=1 then HHcat=1 ; /*single*/
else if nonrelatehh=1 then HHcat=4; /* non relate households*/ 
end;

else if hhtype in (1,2,3) then do; /*family household*/
    if hhtype=1 & numprec=2 then HHcat=2 ; /*couple without kid*/
else HHcat=3; /*other family*/
end;


if hhtype in (0, 9) then HHcat=.;

else HHcat=5;
‘’’

maggiesu0725 replied:
The else HHcat=5 overwrites all the other categories assigned. I had to modify the code a bit to :
if hhtype in (4,5,6,7) then do; /_non family_/
if numprec=1 then HHcat=1 ; /_single_/
else if nonrelatehh=1 then HHcat=4; /* non relate households*/
end;

else if hhtype in (1,2,3) then do; /_family household_/
if hhtype=1 & numprec=2 then HHcat=2 ; /_couple without kid_/
else HHcat=3; /_other family_/
end;

else if hhtype in (0, 9) then do;
 HHcat=.;
end;
else HHcat=5;

But now no 5 is assigned....

rpitingolo replied:
I think you want 
else if hhtype in (0, 9) then do;
HHcat=5;
end;

Where 5 is NA/undetermined. The final else statement doesn't do anything because all the values for hhtype are already accounted for. 
hhtype
hhtype

ptatian replied:
@rpitingolo @maggiesu0725 As a follow up on the discussion of household types, part of the problem was relying upon the IPUMS HHTYPE var. Certain types of households got a HHTYPE=9 because IPUMS couldn't determine how they fit into their classification but they were still households that we want to count. Also, I found a few mistakes in the HHTYPE coding. In the end, I changed from using HHTYPE to creating our own variables to classify the households. This also allowed more consistent coding with 2000, where we did not have HHTYPE.

maggiesu0725 commented on Prog/Demographic trends.sas:
```diff
@@ -0,0 +1,1436 @@
+data COGSvacant_2017 (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309" ...
+set Ipums.Acs_2017_vacant_dc Ipums.Acs_2017_vacant_md Ipums.Acs_2017_vacant_va;
+
+	%assign_jurisdiction; 
```
I can only be used for 2017 IPUMS because that's just how the macro was written for. I could write separate macro for 2000 PUMA and county files.

ptatian replied:
@maggiesu0725 OK, don't worry about it, then.

maggiesu0725 commented:
I've addressed most of Peter's comments. The only one pending is the last comment, where I assign HH categories. I modified the code but now can't get HHcat =5 to show up in my data. The datastep is : data hhtype_2010. Could you take a look? @rpitingolo @ptatian

ptatian commented:
@maggiesu0725 @rpitingolo OK, I will take a look.

ptatian commented:
@lhendey I think I asked this before but if we are using 2016 AMI family of 4 income ranges for summarizing household income, then should we be adjusting all incomes to 2016 dollars? @rpitingolo @maggiesu0725

lhendey commented:
@ptatian 
sounds right to me, my gaps/targets piece is using 2016 dollars for everything. @maggiesu0725 
@rpitingolo

ptatian commented:
@lhendey Thank you. I will make that adjustment. @maggiesu0725 @rpitingolo

ptatian commented:
@maggiesu0725 @rpitingolo I think this is close to being finished, but I want to have a chance to review the data before we finalize.

rpitingolo commented:
@ptatian thanks for the update. @maggiesu0725

ptatian commented:
@rpitingolo @maggiesu0725 OK to merge and do final batch submit for the demographic data.
